src/server/isomorphic.ts

```diff
@@ -80,16 +80,17 @@ export default (context: IServerContext) => {
 
     router
     .onReady(() => {
-      const matchedComponents: Component[] = [App as Component].concat(router.getMatchedComponents());
-
       if (router.currentRoute.fullPath !== context.url) {
         return reject({ code: 302, path: router.currentRoute.fullPath });
       }
 
-      if (matchedComponents.length === 1) {
+      const routeMatchesCatchAll = router.currentRoute.matched.some((match) => match.path === '*');
+      if (routeMatchesCatchAll) {
```

Comment:
this could be really tricky if you have to implement a CMS like contentful. Could you write some comments that this is the part that has to be changed as soon as you want to use the CatchAll route for something else than not-found?